src/components/ArtifactTypeSelect.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <p-select v-model="model" :options="options" class="artifact-type-select" />
+</template>
+
+<script lang="ts" setup>
+  import { SelectOption } from '@prefecthq/prefect-design'
+  import { computed } from 'vue'
+  import { ArtifactType, artifactTypes } from '@/models'
+  import { capitalize } from '@/utilities'
+
+  const props = defineProps<{
+    selected: string | null,
+  }>()
+
+  const emit = defineEmits<{
+    (event: 'update:selected', value: string | null): void,
+  }>()
+
+  const model = computed({
+    get() {
+      return props.selected
+    },
+    set(value: string | null) {
+      emit('update:selected', value)
+    },
+  })
+
+  const options: (SelectOption & { value: ArtifactType | null })[] = [
+    { label: 'All', value: null },
+    ...artifactTypes.map((type) => ({ label: capitalize(type), value: type })),
+  ].sort((alpha, beta) => alpha.label.localeCompare(beta.label))
```

Comment:
super nit: this is almost code golf lol. Its a super small component so its fine. But if you wanted to I would break this out a bit.

Reply:
when all i really wanted is:
```ts
const options: (SelectOption & { value: ArtifactType | null })[] = [
    { label: 'All', value: null },
    { label: 'Markdown', value: 'markdown' },
    { label: 'Table', value: 'table' },
  ]
```

Reply:
That's so much cleaner haha. But I'm not sure it'll guarantee that if a new artifact type is added it'll get added here. So that's worth looking into